controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -307,10 +309,40 @@ public String getScopeName() {
                             long scaleTs = record.getLeft();
                             CompletableFuture<List<Segment>> list = Futures.allOfWithResults(
                                     record.getRight().stream().map(this::getSegment)
-                                    .collect(Collectors.toList()));
+                                            .collect(Collectors.toList()));
+
+                            return list.thenApply(segments -> new ImmutablePair<>(scaleTs, segments));
+                        }).collect(Collectors.toList())))
+                .thenApply(scalePair -> {
+                    final AtomicReference<List<Segment>> previous = new AtomicReference<>();
+                    return scalePair.stream().sorted(Comparator.comparingLong(ImmutablePair::getLeft))
+                            .map(pair -> {
+                                long splits = 0;
+                                long merges = 0;
+                                if (previous.get() != null) {
```

Comment:
I vaguely remember the discussion around ascending and descending order. Could you remind me of the context so that I understand why we are getting rid of it here?

Reply:
I dont know about the discussion around ascending vs descending.. @jkhalack mentioned that the response that is ultimately sent back to UI from the api has a convoluted order.. 
a reference event (the scaling event right before the requested time interval) is added just before this list and this list used to be in descending order
so i have  made it all  ascending as that works for them.